crates/ty_server/src/server/api/traits.rs

```diff
@@ -56,6 +100,10 @@ pub(super) trait BackgroundDocumentRequestHandler: RetriableRequestHandler {
 }
 
 /// A request handler that can be run on a background thread.
+///
+/// Unlike [`BackgroundDocumentRequestHandler`], this handler operates on the entire workspace
+/// without being tied to a specific document. It is useful for operations that require access to
```

Comment:
Same here. You may want to reprhase this to *on the entire session* if you decide to rename `WorkspaceSnapshot` to `SessionSnapshot`